### PR TITLE
chore/deps: upgrade alertmanager package dependency, improve logs and tests

### PR DESCRIPTION
**Pull request by bobheadxi**
Upgrades to our forked update v0.27.0, which matches the Alertmanager version we deploy: https://github.com/sourcegraph/alertmanager/commit/3695ef8bcc9a5bf22945c43067b092e79644f1f3. Upon closer inspection I also realized I upgraded `prometheus/common` too far in https://github.com/sourcegraph/sourcegraph/pull/63328 - I've downgraded it to match the revision of Alertmanager we are using, while _also_ fulfilling the OpenFGA dependency https://github.com/sourcegraph/sourcegraph/pull/63329#discussion_r1646630946 for https://github.com/sourcegraph/sourcegraph/pull/63173 💀

The latest version of `prometheus/common` marshals configuration values that are unknown to our version of Alertmanager (v0.27.0) which rejects the generated configuration from `prom-wrapper`.

I've also made a few updates to improve the testing and improve the prometheus and alertmanager output by forwarding them to differently-scoped loggers and crude conversion of the log levels:

image

Related: https://github.com/sourcegraph/sourcegraph/pull/63171
Closes CORE-186

## Test plan

`sg start` and `sg run prometheus`, update some alerting configs in http://localhost:9090/alertmanager/#/status:

In personal settings:

```json
{
  "alerts.hideObservabilitySiteAlerts": false
}
```

No banners show up indicating Prometheus is unhealthy.

### CODE REVIEW

**Comment by bobheadxi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63329?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#63329** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63329?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#63328** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63328?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bobheadxi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by Strum355 — APPROVED** [review]

File: cmd/executor/internal/apiclient/queue/client.go [hunk 1]
```diff
@@ -321,7 +321,7 @@ func gatherMetrics(logger log.Logger, gatherer prometheus.Gatherer) (string, err
 		return "", err
 	}
 	var buf bytes.Buffer
-	enc := expfmt.NewEncoder(&buf, expfmt.NewFormat(expfmt.TypeTextPlain))
+	enc := expfmt.NewEncoder(&buf, expfmt.FmtText)
```
Comment (unknwon) [inline cmd/executor/internal/apiclient/queue/client.go:324]: Revert `common` version to continue using `expfmt.FmtText` would unsatifiy OpenFGA's requirement of `common` 💀

Comment (bobheadxi) [reply cmd/executor/internal/apiclient/queue/client.go]: Upgraded to the version declared by OpenFGA, v0.48.0 https://github.com/openfga/openfga/blob/77c666d5063c51216f7ffc0b6cc8a0e62957b241/go.mod#L106 - thankfully looks like this is still compatible, running some tests...

Comment (bobheadxi) [reply cmd/executor/internal/apiclient/queue/client.go]: Okay, I think this is good to go

**Pull request merged**